exporter/elasticsearchexporter/elasticsearch_bulk.go

```diff
@@ -132,27 +131,6 @@ func newElasticsearchClient(logger *zap.Logger, config *Config) (*esClientCurren
 		Logger:            (*clientLogger)(logger),
 	})
 }
-func newRetryOnErrorFunc(retryDisabled bool) func(_ *http.Request, err error) bool {
-	if retryDisabled {
-		return func(_ *http.Request, err error) bool {
-			return false
-		}
-	}
-
-	return func(_ *http.Request, err error) bool {
-		var netError net.Error
-		shouldRetry := false
-
-		if isNetError := errors.As(err, &netError); isNetError && netError != nil {
-			// on Timeout (Proposal: predefined configuratble rules)
-			if !netError.Timeout() {
-				shouldRetry = true
-			}
-		}
-
-		return shouldRetry
-	}
-}
```

Comment:
Can I ask why this is being removed from this file?

Reply:
This function is solely used for the client option field "retryOnError" which exists in v8

Reply:
Your change sets has the imports version 7, wouldn't that mean that this function is still needed?

Reply:
No. as the purpose of this PR is to roll back to v7. this function cannot be used anywhere in the client, as it is dedicatedly written for "retryOnError" callback field of v8. so it is better to remove it.